### ISSUE TITLE
Incorrect validation on keysend route hints

### ISSUE THREAD

**Issue by Evanfeenstra** [ElementsProject/lightning#6143]
### Issue and Steps to Reproduce

The `param_routehint_array` function here https://github.com/ElementsProject/lightning/blob/master/common/json_param.c#L953 checks if the RoutehintList is an array (JSMN_ARRAY). However the RoutehintList is actually an object that has the array of hints inside a `hints` field https://github.com/ElementsProject/lightning/blob/master/cln-grpc/proto/primitives.proto#L73. The `param_routehint` function looks like it has the same problem.

You can try it using a `cln-grpc` client, like in this example https://github.com/Evanfeenstra/cln-grpc-routehints-test/blob/master/src/cln.rs#L50. The exact error message is:
```
Routehint array routehints (\\"{\\"hints\\":[{\\"hops\\":[{\\"expirydelta\\":40,\\"feebase\\":\\"1000msat\\",\\"feeprop\\":1,\\"id\\":\\"02c7046d20f62012362ccf835fe5b4d4a1708e518592f216afeefabeadfc20154b\\",\\"scid\\":\\"1x5x1\\"}]}]}\\") is not an array
```

**Comment by cdecker** [comment]
Hm, that's definitely wrong on my side. Seems like this is the only case of a 2D array, which is treated a bit differently in the code generation tool.

 - [ ] Add a custom conversion for routehints if not already present
 - [ ] Amend the `msggen` to either not handle 2D arrays or fix its behavior
 - [ ] Publish a new version to crates.io

However I am going to be offline till end of month, hopefully that is not too late :-)

**Comment by cdecker** [comment]
Hi @Evanfeenstra, I pushed a preliminary fix as #6154. It converts the custom types correctly to the 2d array now. For cln-grpc the inverse conversion is not required but we may want it at some time in the future, hence why there are the `todo!()` markers. I'll push the new version to crate.io as soon as the PR is merged ^^

**Comment by Evanfeenstra** [comment]
thanks for working on it @cdecker !
